backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java

```diff
@@ -92,18 +96,43 @@
   private ProjectDAO projectDAO;
   private ExperimentRunDAO experimentRunDAO;
   private ArtifactStoreDAO artifactStoreDAO;
+  private final AuditLogLocalDAO auditLogLocalDAO;
+  private static final String SERVICE_NAME =
+      String.format("%s.%s", ModelDBConstants.SERVICE_NAME, ModelDBConstants.PROJECT);
 
   public ProjectServiceImpl(
       AuthService authService,
       RoleService roleService,
       ProjectDAO projectDAO,
       ExperimentRunDAO experimentRunDAO,
-      ArtifactStoreDAO artifactStoreDAO) {
+      ArtifactStoreDAO artifactStoreDAO,
+      AuditLogLocalDAO auditLogLocalDAO) {
     this.authService = authService;
     this.roleService = roleService;
     this.projectDAO = projectDAO;
     this.experimentRunDAO = experimentRunDAO;
     this.artifactStoreDAO = artifactStoreDAO;
+    this.auditLogLocalDAO = auditLogLocalDAO;
+  }
+
+  private void saveAuditLogs(
+      UserInfo userInfo, String action, List<String> resourceIds, String metadataBlob) {
+    List<AuditLogLocalEntity> auditLogLocalEntities = new ArrayList<>();
+    resourceIds.forEach(
+        resourceId ->
+            auditLogLocalEntities.add(
+                new AuditLogLocalEntity(
+                    SERVICE_NAME,
+                    authService.getVertaIdFromUserInfo(
+                        userInfo == null ? authService.getCurrentLoginUserInfo() : userInfo),
+                    action,
+                    resourceId,
+                    ModelDBConstants.PROJECT,
+                    Service.MODELDB_SERVICE.name(),
+                    metadataBlob)));
```

Comment:
This can be written more concisely:

List<AuditLogLocalEntity> auditLogLocalEntities = resourceIds.stream().map(resourceId -> new 
              AuditLogLocalEntity(
                    SERVICE_NAME,
                    authService.getVertaIdFromUserInfo(
                        userInfo == null ? authService.getCurrentLoginUserInfo() : userInfo),
                    action,
                    resourceId,
                    ModelDBConstants.PROJECT,
                    Service.MODELDB_SERVICE.name(),
                    metadataBlob)
             .collect(Collectors.toList());

Reply:
Resolved